examples/bert/run_glue_finetuning.py

```diff
@@ -215,15 +230,29 @@ def preprocess_data(inputs, labels):
         hidden_size=bert_config["hidden_size"],
         num_classes=3 if FLAGS.task_name in ("mnli", "ax") else 2,
     )
-    finetuning_model.compile(
-        optimizer=keras.optimizers.Adam(learning_rate=FLAGS.learning_rate),
-        loss="sparse_categorical_crossentropy",
-        metrics=["accuracy"],
-    )
-    finetuning_model.fit(
-        train_ds, epochs=FLAGS.epochs, validation_data=validation_ds
+
+    hypermodel = BertHyperModel(finetuning_model)
+
+    # Initialize the random search over the 4 learning rate parameters, for 4
+    # trials and 3 epochs for each trial.
+    tuner = kt.RandomSearch(
+        hypermodel=hypermodel,
+        objective="val_accuracy",
+        max_trials=4,
+        overwrite=True,
+        directory="tuning_hp_bert",
+        project_name="glue_finetuning_hp",
     )
 
+    tuner.search(train_ds, epochs=3, validation_data=validation_ds)
+
+    # Extract the best hyperparameters after the search.
+    best_hp = tuner.get_best_hyperparameters()[0]
+    finetuning_model = hypermodel.build(best_hp)
+
+    print("Training the model using the best hyperparameters found.")
```

Comment:
Could we update this to print the actual learning rate and other parameters we used?

Reply:
Yeah, I'll add another print statement to do output the learning rate.